### PR TITLE
feat: add co-editor settings

### PR DESCRIPTION
**Pull request by louisewang1**
Add co-editor display setting in publish tab

### CODE REVIEW

**Comment by louisewang1** [comment]
image

File: src/components/item/publish/CoEditorSettings.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+const CoEditorSettings = ({ item }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const { mutate: updateDisplayCoEditors } = useMutation(EDIT_ITEM);
+
+  // user
+  const { isLoading: isMemberLoading } = useContext(CurrentUserContext);
+
+  // current item
+  const { itemId } = useParams();
```
Comment (pyphilia) [inline src/components/item/publish/CoEditorSettings.js:39]: You can get the item id from the item as well, for consistency.

File: src/components/item/publish/CoEditorSettings.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  };
+
+  return (
+    <>
+      <Typography variant="h6" className={classes.title}>
+        {t('Co-Editors')}
+      </Typography>
+      <Typography variant="body1">
+        {t(
+          'Do you want to display co-editors after published? All users with edit permission will be displayed.',
```
Comment (pyphilia) [inline src/components/item/publish/CoEditorSettings.js:74]: `Do you want to display co-editors on the publication page? All users with edit or admin permissions will be displayed.`

File: src/components/item/publish/CoEditorSettings.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      >
+        <FormControlLabel
+          value={DISPLAY_CO_EDITOR_OPTIONS.YES.value}
+          control={<Radio color="primary" />}
+          label={t(DISPLAY_CO_EDITOR_OPTIONS.YES.label)}
+        />
+        <FormControlLabel
+          value={DISPLAY_CO_EDITOR_OPTIONS.NO.value}
+          control={<Radio color="primary" />}
+          label={t(DISPLAY_CO_EDITOR_OPTIONS.NO.label)}
```
Comment (pyphilia) [inline src/components/item/publish/CoEditorSettings.js:91]: I think we can go with a `Object.values(DISPLAY_CO_EDITOR_OPTIONS).map(...)`

File: src/config/constants.js [hunk 4]
```diff
@@ -249,3 +249,14 @@ export const BUTTON_TYPES = {
 
 export const MEMBERSHIP_TABLE_HEIGHT = 400;
 export const MEMBERSHIP_TABLE_ROW_HEIGHT = 75;
+
+export const DISPLAY_CO_EDITOR_OPTIONS = {
```
Comment (pyphilia) [inline src/config/constants.js:253]: `DISPLAY_CO_EDITORS_OPTIONS`

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the PR! I've left some comments :
Could we add trivial tests? Check the setting is displayed, and you can click it and it trigger the correct request.

**Comment by louisewang1** [comment]
checkbox for email notification
image

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the PR! I've left some minor comments 🍇

**Pull request merged**